lib/cast.js

```diff
@@ -260,6 +260,11 @@ module.exports = function cast(schema, obj, options, context) {
           continue;
         }
 
+        // allows finding nested discriminators (gh-12318)
+        if (schema.discriminators && Object.keys(schema.discriminators).length > 0) {
```

Comment:
As far as I can tell, this check basically tells Mongoose to ignore strict mode if there are discriminators. I don't think that's the problem we're trying to solve here.

Reply:
I see, thanks for the feedback. I also tried putting this a little below, wrapping L283.
https://github.com/Automattic/mongoose/blob/6f8d35b40285a83c6646e2b39e3df601a7912c57/lib/cast.js#L282-L284
This also fixes the querying issue, but as far as I remember, it breaks a few tests. And it seemed wrong to put it there.

Reply:
I saw Fabian's reply on the mentioned issue. Should we do this on schema level or just find another way to solve it?

Reply:
Yeah I think the more correct approach is setting the `strictQuery` option. Ignoring strict mode for discriminators isn't something we're going to do. I'm going to close this PR, but thanks for your hard work on this :+1: